### PR TITLE
feat(workflow_test_release): add workflow_test_release job DSL

### PR DESCRIPTION
**Pull request by vdice**
To be used for testing a workflow chart associated with a given release.

TODO after LGTMs:
- [x] update https://ci.deis.io/job/workflow-test-seed-job/ to build this job as well

### CODE REVIEW

**Comment by vdice** [comment]
Live example of this job exists in https://ci.deis.io/job/workflow-test-release/2/ which was built straight from the DSL on this branch...

File: jobs/workflow_test.groovy [hunk 1]
```diff
@@ -131,7 +129,6 @@ import utilities.StatusUpdater
 
     environmentVariables {
       env('COMMIT', isMaster)
-      env('PARALLEL_TESTS', true)
```
Comment (bacongobbler) [inline jobs/workflow_test.groovy:134]: how come we're removing the parallel tests envvar here?

Comment (vdice) [inline jobs/workflow_test.groovy:134]: Good question.  This was a temporary env var used when we had two e2e charts, one for running tests in a serialized fashion, the other concurrently.  We now just have one e2e chart, defaulting to running in parallel (using multiple `GINKGO_NODES`).  To run in serial using the same chart, we can set `GINKGO_NODES` to `1`.

Comment (bacongobbler) [inline jobs/workflow_test.groovy:134]: Right! I forgot about that.

**Pull request merged**